receiver/azureeventhubreceiver/eventhubhandler.go

```diff
@@ -43,24 +45,28 @@ type listerHandleWrapper interface {
 }
 
 type eventhubHandler struct {
-	hub          hubWrapper
-	dataConsumer dataConsumer
-	config       *Config
-	settings     receiver.Settings
-	cancel       context.CancelFunc
+	hub           hubWrapper
+	dataConsumer  dataConsumer
+	config        *Config
+	settings      receiver.Settings
+	cancel        context.CancelFunc
+	storageClient storage.Client
 }
 
 func (h *eventhubHandler) run(ctx context.Context, host component.Host) error {
 	ctx, h.cancel = context.WithCancel(ctx)
 
-	storageClient, err := adapter.GetStorageClient(ctx, host, h.config.StorageID, h.settings.ID)
-	if err != nil {
-		h.settings.Logger.Debug("Error connecting to Storage", zap.Error(err))
-		return err
+	if h.storageClient == nil { // set manually for testing.
```

Comment:
I'm not going to block this PR since it fixes an important bug, but generally I'm not a fan of tests relying on special access to the internals of the code. A better way to do this would be to mock the host, so that it returns the mock client.